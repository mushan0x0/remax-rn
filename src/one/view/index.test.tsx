import TestRenderer from 'react-test-renderer';
import View from './index';
import React from 'react';

jest.useFakeTimers();

describe('View', () => {
  it('基本渲染', () => {
    expect(
      TestRenderer.create(
        <View style={{ lineHeight: 1.1, background: 'red', margin: '20px 0' }}>
          <View
            style={{
              fontSize: 14,
              fontWeight: 400,
              color: 'red',
              border: '1px solid #eee',
              borderRadius: '50%',
              lineHeight: '2',
              padding: '20px 0',
            }}
          >
            <View style={{ width: '2em', lineHeight: 16 }}>233</View>
            <View style={{ width: '2em', lineHeight: 1 }}>233</View>
          </View>
        </View>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
