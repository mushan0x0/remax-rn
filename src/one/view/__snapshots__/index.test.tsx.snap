// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View 基本渲染 1`] = `
<View
  activeOpacity={0.5}
  className="view"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "borderRadius": undefined,
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "nowrap",
      "lineHeight": 19.360000000000003,
      "marginBottom": 39.999975,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 39.999975,
      "opacity": undefined,
      "position": undefined,
    }
  }
>
  <View
    activeOpacity={0.5}
    className="view"
    onPress={[Function]}
    style={
      Object {
        "border": "1px solid #eee",
        "borderColor": "#eee",
        "borderRadius": 375,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "red",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 30.800000000000004,
        "opacity": undefined,
        "paddingBottom": 39.999975,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 39.999975,
        "position": undefined,
      }
    }
  >
    <View
      activeOpacity={0.5}
      className="view"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "lineHeight": 16,
          "opacity": undefined,
          "position": undefined,
          "width": 57.2,
        }
      }
    >
      <Text
        className="view"
        style={
          Object {
            "color": "red",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 16,
            "paddingTop": 2.8000000000000003,
            "textAlign": undefined,
            "textAlignVertical": "center",
          }
        }
      >
        233
      </Text>
    </View>
    <View
      activeOpacity={0.5}
      className="view"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "lineHeight": 14,
          "opacity": undefined,
          "position": undefined,
          "width": 57.2,
        }
      }
    >
      <Text
        className="view"
        style={
          Object {
            "color": "red",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 14,
            "paddingTop": 2.8000000000000003,
            "textAlign": undefined,
            "textAlignVertical": "center",
          }
        }
      >
        233
      </Text>
    </View>
  </View>
</View>
`;
